### ISSUE TITLE
Output GaussianMixture in sigma_point::unscented_transform does not reflect the internal composition of the measurement vector

### ISSUE THREAD

**Issue by xEnVrE** [robotology/bayes-filters-lib#65]
In `sigma_point::unscented_transform`, the output `GaussianMixture` is initialized considering the whole row size of the propagated sigma points.
https://github.com/robotology/bayes-filters-lib/blob/9b623d04640e94df48e67a8279883b074ea21bd8/src/BayesFilters/src/sigma_point.cpp#L116

Since, however, we know the internal composition of the measurement sizes

https://github.com/robotology/bayes-filters-lib/blob/9b623d04640e94df48e67a8279883b074ea21bd8/src/BayesFilters/src/sigma_point.cpp#L105-L106

we should probably have
```
 GaussianMixture output(input.components, output_size.first, output_size.second); 
```

The current implementation does not suffer from this bug, since what is important is that the means are evaluated taking into account the `linear` and `circular` sizes 

https://github.com/robotology/bayes-filters-lib/blob/9b623d04640e94df48e67a8279883b074ea21bd8/src/BayesFilters/src/sigma_point.cpp#L129-L130

However, as a user, I would probably like to know the correct sizes directly from the output `GaussianMixture` returned by the method, using `mixture.dim_linear` and `mixture.dim_circular`.

**Comment by claudiofantacci** [comment]
That's right! 👍 

Could you please submit a PR for this?

**Comment by xEnVrE** [comment]
Yes, I will.

**Comment by claudiofantacci** [comment]
Closed by #73.
